doc/devel/development_setup.rst

```diff
@@ -0,0 +1,135 @@
+.. _installing_for_devs:
+
+=====================================
+Setting up Matplotlib for development
+=====================================
+
+.. _dev-environment:
+
+Creating a dedicated environment
+================================
+You should set up a dedicated `virtual environment`_ to decouple your
+Matplotlib development from other Python and Matplotlib installations on your
+system. This is the environment you can safely develop and test your changes
+in.
```

Comment:
maybe add a sentence like "Here we recommend the `venv` virtual environment, you may use others such as conda." (especially since you wrote the env file!)

Reply:
Yes. More on that will come with #17096.